src/app/components/modal/modal.ts

```diff
@@ -27,7 +27,22 @@ import { Tag, Preset, PrimaryTag, TagConfig, OsmGoFeature } from '../../../type'
 
 import { cloneDeep, isEqual, findIndex } from 'lodash'
 import { TranslateService } from '@ngx-translate/core'
-
+import {
+    LineString,
+    MultiLineString,
+    MultiPolygon,
+    Point,
+    Polygon,
+} from 'geojson'
+
+export interface ModalDismissData {
+    redraw?: boolean
+    type?: string
+    geojson?: OsmGoFeature<
+        Point | LineString | MultiLineString | Polygon | MultiPolygon
+    >
```

Comment:
I'm not entirely sure about the potential geometries here. I think at least `Point` is clear and I guess that partially the app allows to also add other geometries such as `LineString` or `Polygon`. Please review this part carefully.